di-ipv-ticf-stub/lambdas/src/management/services/userEvidenceService.ts

```diff
@@ -0,0 +1,111 @@
+import { config } from "../../common/config";
+
+const dynamoClient = config.isLocalDev
+  ? new DynamoDB({
+      endpoint: config.localDynamoDbEndpoint,
+      region: config.region,
+    })
+  : new DynamoDB({ region: config.region });
+
+export async function persistUserEvidence(
+  userId: string,
+  ticfEvidenceItemReq: TicfEvidenceItem
+): Promise<void> {
+  const ticfStubTtl = getTtl(
+    await getSsmParameter(config.ticfParamBasePath + "ticfStubTtl")
+  );
+
+  const existingUserEvidence = await getUserEvidence(userId);
+  if (!existingUserEvidence) {
+    const userEvidence: UserEvidenceItem = {
+      userId: userId,
+      evidence: ticfEvidenceItemReq,
+      ttl: ticfStubTtl,
+    };
+    await saveUserEvidence(userEvidence);
+  } else {
+    await updateUserEvidence(userId, ticfEvidenceItemReq, ticfStubTtl);
+  }
+}
+
+export async function getUserEvidence(
+  userId: string
+): Promise<UserEvidenceItem | null> {
+  console.info(`Get user record.`);
+  const getItemInput: GetItemInput = {
+    TableName: config.ticfStubUserEvidenceTableName,
+    Key: marshall({
+      userId: userId,
+    }),
+  };
+  const { Item } = await dynamoClient.getItem(getItemInput);
+  const userEvidenceItem = Item ? (unmarshall(Item) as UserEvidenceItem) : null;
+  if (userEvidenceItem === null) {
+    console.info(`Record not found for user.`);
+  }
+  return userEvidenceItem;
+}
+
+async function saveUserEvidence(userEvidence: UserEvidenceItem) {
+  console.info(`Save user record.`);
+  const putItemInput: PutItemInput = {
+    TableName: config.ticfStubUserEvidenceTableName,
+    Item: marshall(userEvidence),
+  };
+  const { Attributes } = await dynamoClient.putItem(putItemInput);
+  const userEvidenceItem = Attributes
+    ? (unmarshall(Attributes) as UserEvidenceItem)
+    : null;
+  if (userEvidenceItem === null) {
+    console.info(`User record not saved.`);
+  }
+  return userEvidenceItem;
+}
+
+async function updateUserEvidence(
+  userId: string,
+  updatedEvidence: TicfEvidenceItem,
+  ticfStubTtl: number
+) {
+  console.info(`Update user record.`);
+  const updateItemInput: UpdateItemInput = {
+    TableName: config.ticfStubUserEvidenceTableName,
+    Key: marshall({
+      userId: userId,
+    }),
+    UpdateExpression: "set evidence = :updatedEvidence, #attrTtl = :updatedTtl",
+    ExpressionAttributeNames: { "#attrTtl": "ttl" },
+    ExpressionAttributeValues: marshall({
+      ":updatedEvidence": updatedEvidence,
+      ":updatedTtl": ticfStubTtl,
+    }),
+    ReturnValues: "ALL_NEW",
+  };
+  const { Attributes } = await dynamoClient.updateItem(updateItemInput);
+  const userEvidenceItem = Attributes
+    ? (unmarshall(Attributes) as UserEvidenceItem)
+    : null;
+  if (userEvidenceItem === null) {
+    console.info(`User record not updated.`);
+  }
+  return userEvidenceItem;
+}
+
+function getTtl(ticfStubTtl: string): number {
+  const newLocal = new Date();
+  newLocal.setSeconds(parseInt(ticfStubTtl));
```

Comment:
I don't think this is quite what you want - this will set the 'seconds' value of the date to the TTL value. I.e. if the date is `2023-12-01T11:23:45` and the TTL is 30 seconds it'll be set to `2023-12-01T11:23:30`.

Instead, I think you just want to add the TTL to the time:
```
return Math.floor(Date.now() / 1000) + ticfStubTtl
```

Reply:
seems working ok 
Added record at 1/12/2023 @ 12:50
ttl - 1 December 2023 13:19:00

Reply:
That's weird - are you sure?
```
> const date = new Date('2023-12-01T11:23:45')
undefined
> date
2023-12-01T11:23:45.000Z
> date.setSeconds(30)
1701429810000
> date
2023-12-01T11:23:30.000Z
```

Reply:
its configurable - 1800

Reply:
Ok, because it's a large number it's _sort of_ working, but it's still not really the right method to use.

It's actually setting the seconds value to 1800, but because seconds only goes to 60, it's overflowing and setting the seconds value to zero but adding 30 minutes. In practice this means you'll add between 29-30 minutes, rounding down to the nearest minute.

Arguably that's fine for our purposes, but I'd still prefer to just add the TTL on properly!

Reply:
Done